### ISSUE TITLE
args: Warning when local module use a local module_utils

### ISSUE THREAD

**Issue by baierjan** [ansible/ansible-lint#3891]
This is a reproducer for #2835 which is still present, I believe the patch from https://github.com/ansible/ansible-lint/issues/2835#issuecomment-1354737820 might still be relevant.

##### Summary

Unexpected warning when local module use a local module_utils. I believe `ansible-lint` is not able to load module_utils properly when `ANSIBLE_MODULE_UTILS` is set to custom path (i.e. when local module_utils are used).

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.22.0 using ansible-core:2.15.6 ansible-compat:4.1.10 ruamel-yaml:0.17.35 ruamel-yaml-clib:0.2.8
```

- ansible installation method: OS package (openSUSE Tumbleweed 2.15.6-1.1)
- ansible-lint installation method: OS package (openSUSE Tumbleweed 6.22.0-1.1)

##### STEPS TO REPRODUCE

1. Lets have a simple Ansible repository containing the following files.

```console
#  tree .
.
├── book.yaml
└── plugins
    ├── modules
    │   └── test_module.py
    └── module_utils
        └── my_test.py

4 directories, 3 files
```

```yaml
# cat book.yaml
---
- name: Example
  hosts: localhost

  tasks:
    - name: Example task
      test_module:
```

```py
# cat plugins/modules/test_module.py
#!/usr/bin/python

from ansible.module_utils.basic import AnsibleModule
from ansible.module_utils.my_test import hello

def main():
    module = AnsibleModule(argument_spec=dict(), supports_check_mode=True)
    module.exit_json(message=hello())

if __name__ == '__main__':
    main()
```

```py
# cat plugins/module_utils/my_test.py
def hello():
    return "Hello"
```

2. This is a perfectly valid for Ansible (note: _ansible.cfg_ is empty, i.e. all defaults)

```console
# ANSIBLE_LIBRARY=./plugins/modules ANSIBLE_MODULE_UTILS=./plugins/module_utils ansible-playbook -v book.yaml
Using /etc/ansible/ansible.cfg as config file
[WARNING]: provided hosts list is empty, only localhost is available. Note that the
implicit localhost does not match 'all'

PLAY [Example] *************************************************************************

TASK [Gathering Facts] *****************************************************************
ok: [localhost]

TASK [Example task] ********************************************************************
ok: [localhost] => {"changed": false, "message": "Hello"}

PLAY RECAP *****************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```

3. But ansible-lint has problem to load the my_test.py module utils.

```console
# ANSIBLE_LIBRARY=./plugins/modules ANSIBLE_MODULE_UTILS=./plugins/module_utils ansible-lint
WARNING  Ignored exception from ArgsRule.<bound method AnsibleLintRule.matchtasks of args: Validating module arguments.> while processing book.yaml (playbook): No module named 'ansible.module_utils.my_test'

Passed: 0 failure(s), 0 warning(s) on 3 files. Last profile that met the validation criteria was 'production'.
```

##### Desired Behavior

No error/warning about ignoring exception while importing local module_utils, the module `ansible.module_utils.my_test` is loaded properly.

##### Actual Behavior

The module is not loaded, maybe the `ANSIBLE_MODULE_UTILS` path is not taken into account?

```console
# ANSIBLE_LIBRARY=./plugins/modules ANSIBLE_MODULE_UTILS=./plugins/module_utils ansible-lint -v book.yaml`
INFO     Identified / as project root due file system root.
INFO     Set ANSIBLE_LIBRARY=/home/user/.cache/ansible-compat/1a6f0d/modules:plugins/modules:/home/user/Projects/ansible-lint/plugins/modules:./plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/home/user/.cache/ansible-compat/1a6f0d/collections:/home/user/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/home/user/.cache/ansible-compat/1a6f0d/roles:/home/user/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
INFO     Set ANSIBLE_LIBRARY=/home/user/.cache/ansible-compat/1a6f0d/modules:plugins/modules:/home/user/Projects/ansible-lint/plugins/modules:./plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/home/user/.cache/ansible-compat/1a6f0d/collections:/home/user/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/home/user/.cache/ansible-compat/1a6f0d/roles:/home/user/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
INFO     Executing syntax check on playbook book.yaml (0.38s)
WARNING  Ignored exception from ArgsRule.<bound method AnsibleLintRule.matchtasks of args: Validating module arguments.> while processing book.yaml (playbook): No module named 'ansible.module_utils.my_test'
```

**Comment by ssbarnea** [comment]
This looks like a good idea, a pull-requests would be welcomed.

**Comment by ssbarnea** [comment]
I tested with code from main branch and I was not able reproduce the bug. The only quirk is that you will stills see one warning from ArgsRule.matchtasks rule, but linting will pass.

**Comment by baierjan** [comment]
Hi @ssbarnea, I just tried that with a newer version and the problem is still there. 

I am not sure what do you mean by not able to reproduce the bug when in your comment you clearly states that the unwanted warning appeared. That's exactly what is this bug about:

```
# ansible-lint --version
ansible-lint 24.6.0 using ansible-core:2.16.7 ansible-compat:24.6.0 ruamel-yaml:0.18.6 ruamel-yaml-clib:0.2.8
#  ANSIBLE_LIBRARY=./plugins/modules ANSIBLE_MODULE_UTILS=./plugins/module_utils ansible-lint
WARNING  Ignored exception from ArgsRule.matchtasks while processing book.yaml (playbook): No module named 'ansible.module_utils.my_test'

Passed: 0 failure(s), 0 warning(s) on 3 files. Last profile that met the validation criteria was 'production'.
```

**Comment by lae** [comment]
Per above linked PR this issue still exists. Sure, the linter still runs, but you would expect module_utils local to a role would be discoverable and the warning suggests otherwise.

**Comment by baierjan** [comment]
The reproducer in https://github.com/ansible/ansible-lint/issues/2835#issuecomment-2110402878 is using collection, however the reproducer for this issue does not need any collection, it is just a local module with a code in `module_utils`.

> but you would expect module_utils local to a role would be discoverable and the warning suggests otherwise.

It is exactly as you said, no problem with the linting itself (at least I hope) but the warning message is a little bit unexpected.
